### PR TITLE
[tv2hu] Add new extractor

### PR DESCRIPTION
**Pull request by Vukkk**
## Please follow the guide below
- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your _pull request_ (like that [x])
- Use _Preview_ tab to see how your _pull request_ will actually look like

---
### Before submitting a _pull request_ make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
### What is the purpose of your _pull request_?
- [ ] Bug fix
- [x] New extractor
- [ ] New feature

---
### Description of your _pull request_ and other information

Explanation of your _pull request_ in arbitrary form goes here. Please make sure the description explains the purpose and effect of your _pull request_ and is worded well enough to be understood. Provide as much context and examples as possible.

New extractor for hungarian TV2 (tv2.hu)

### CODE REVIEW

File: youtube_dl/extractor/tv2hu.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+            quality = str_to_int(json_data['mp4Labels'][i][:-1])
+
+            formats.append({
+                'protocol': 'http',
+                'url': json_data['bitrates']['mp4'][i],
+                'height': quality,
+                'width': quality/9*16,
+                'ext': 'mp4',
+                'format_id': json_data['mp4Labels'][i],
```
Comment (TRox1972) [inline youtube_dl/extractor/tv2hu.py]: This shouldn't be fatal; use `.get()` instead

File: youtube_dl/extractor/tv2hu.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        manifest_url = json_data['bitrates']['hls']
+
+        formats = self._extract_m3u8_formats(
+            manifest_url, video_id, 'mp4', entry_protocol='m3u8_native')
+
+        for i in range(len(json_data['bitrates']['mp4'])):
+            if json_data['mp4Labels'][i].lower() == 'auto':
+                continue
+
+            quality = str_to_int(json_data['mp4Labels'][i][:-1])
```
Comment (TRox1972) [inline youtube_dl/extractor/tv2hu.py]: Shouldn't be fatal

File: youtube_dl/extractor/tv2hu.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import re
+
+from .common import InfoExtractor
+
+class TV2HUIE(InfoExtractor):
+    IE_NAME = 'tv2.hu'
+    _VALID_URL = r'https?://(?:www\.)?tv2\.hu/(?:musoraink/)?(?P<uploader>[^/]+)/(?:teljes_adasok/)?(?P<id>[0-9]+)_(.+?)\.html'
+    _JSON_URL = r'(?P<json_url>https?://.+?\.tv2\.hu/vod/(?P<upload_date>\d+)/id_(?P<upload_id>\d+).+?&type=json)'
+
+    _TESTS = [{
```
Comment (TRox1972) [inline youtube_dl/extractor/tv2hu.py]: Any reason to have three full tests? If not, you could set two of them to `'only_matching': True`

File: youtube_dl/extractor/tv2hu.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+        return {
+            'id': video_id,
+            'title': self._og_search_title(webpage).strip(),
+            'thumbnail': self._og_search_property('image', webpage),
+            'uploader': self._search_regex(self._VALID_URL, url, 'uploader'),
+            'upload_id': json_url.group('upload_id'),
+            'upload_date': json_url.group('upload_date'),
+            'formats': formats
+        }
```
Comment (TRox1972) [inline youtube_dl/extractor/tv2hu.py]: missing newline

**Comment by remitamine** [comment]
Thanks, Do you want to be credited as one of the authors of youtube-dl? If so under what name?

**Comment by Vukkk** [comment]
No, thank you!

**Pull request closed without merging**